plugins/processors/awsapplicationsignals/internal/normalizer/attributesnormalizer.go

```diff
@@ -65,20 +84,22 @@ func NewAttributesNormalizer(logger *zap.Logger) *attributesNormalizer {
 
 func (n *attributesNormalizer) Process(attributes, resourceAttributes pcommon.Map, isTrace bool) error {
 	n.copyResourceAttributesToAttributes(attributes, resourceAttributes, isTrace)
+	// It's assumed that all attributes are initially inserted as trace attribute, and attributesRenamingForMetric
+	// contains all attributes that will be used for CloudWatch metric dimension. Therefore, we iterate the keys
+	// for enforcing the limits on length.
+	truncateAttributesByLength(attributes)
```

Comment:
Do you need to take into account the `isTrace` flag? `DefaultMetricAttributeLength` implies that it's only for metric attributes and that it shouldn't be applied to trace attributes.

Reply:
Yes, we want to keep metric dimension values and trace annotation values to be the same for correlation.